### DISCUSSION TITLE
Max layout with border_focus

### DISCUSSION THREAD

**Discussion by hamensman** [qtile/qtile#2891] [Show and tell]
I saw @ImAReplicant added support for margins in `Max` layout, which I liked a lot: https://github.com/qtile/qtile/pull/2726
I thought the next logical step would then be to add `border_focus` too.

Functionally redundant since you will only see one window in front of you at a time 😂. But I I wanted the `Max` layout to be aesthetically consistent with the other layouts with the border.

On top of the `margin` parameter added by @ImAReplicant already, I added the `border_focus` **and** `border_normal` parameters.

You may be wondering why the latter, for the same reason why this whole thing was even necessary to begin with. Honestly, it's because when I did a full screen on my (transparent) terminal, I could see that the other window had the `border_focus` around it, which bothered me 😂.

Personally, I've actually written this as a custom function: [custom_max.txt](https://github.com/qtile/qtile/files/7348633/custom_max.txt) (rename to `custom_max.py` accordingly) Learnt that from @elParaguayo!

 Then I wrote `from custom_max import CustomMax` in my config, and also copied the normal `Max` icon into `~/.icons` and renamed _that_ to `layout-custommax.png` accordingly.

![Screenshot_2021-10-14_20-07-47](https://user-images.githubusercontent.com/76445071/137380744-f0e2f194-2089-4294-be28-4784225ee1aa.png)

**Comment by alethiophile** [comment]
Another factor here is that when dealing with multiple screens, you can have a window open in a Max layout that is not focused. In this case it's actually useful to have the border display focus, since application-level display of window focus is inconsistent and often subtle/nonexistent.

**Reply by m-col** [reply]
This is a good point - it does indeed make sense to have both `border_focus` and `border_normal` even if there is only one window ever on that layout :)

**Reply by elParaguayo** [reply]
There's #3030 but, in the interests of getting this done (as that one seems to have died), I can pick this up in a new PR.
